packages/venia-ui/lib/util/formValidators.js

```diff
@@ -32,8 +32,32 @@ export const hasLengthExactly = (value, values, length) => {
     return SUCCESS;
 };
 
+/**
+ * isRequired is provided here for convenience but it is inherently ambiguous and therefore we don't recommend using it.
+ * Consider using more specific validators such as `hasLengthAtLeast` or `mustBeChecked`.
+ */
 export const isRequired = value => {
-    return (value || '').trim() ? SUCCESS : 'The field is required.';
+    const FAILURE = 'The field is required.';
+
+    // If it's a boolean, it must be `true`.
```

Comment:
Is it possible for a checkbox field to hold a value of `undefined`? Is it always `true` or `false`?